web/app/components/base/chat/chat-with-history/hooks.tsx

```diff
@@ -338,11 +346,22 @@ export const useChatWithHistory = (installedAppInfo?: InstalledApp) => {
   }, [isInstalledApp, appId, notify, t, conversationRenaming, originConversationList])
 
   const handleNewConversationCompleted = useCallback((newConversationId: string) => {
+    mutateAppConversationData()
+  }, [mutateAppConversationData, handleConversationIdInfoChange])
+
+  const handleNewConversationStarted = useCallback((newConversationId: string) => {
     setNewConversationId(newConversationId)
+    setConversationChatList(prevConversations => {
+      const {'': _, ...remainingConversations} = prevConversations[appId || '']
+      return {
+        ...prevConversations,
+        [appId || '']: remainingConversations
+      }
+    })
     handleConversationIdInfoChange(newConversationId)
     setShowNewConversationItemInList(false)
     mutateAppConversationData()
-  }, [mutateAppConversationData, handleConversationIdInfoChange])
+  }, [handleConversationIdInfoChange])
```

Comment:
@viswak-cn why have you removed `mutateAppConversationData` from the callback's dependency list?

Reply:
Actually the git's history UI shows like that - in fact it was never added to handleNewConversationStarted's useCallback dependency's list, it was forgotten. Please add it there, it should be good to go.